### PR TITLE
fslrobustfov with padding inorder to remove neck regions in the fsl-bet skull stripped image

### LINKED ISSUE #1928: 🐛 Skull stripping or brain extraction with fsl BET does not properly remove the neck regions in some subjects.
### Describe the bug

When fsl BET is used for skull stripping or brain extraction, the neck regions from the anatomical image does not get removed for some subjects. 
As you can see in this example, the final output of fsl BET(red) includes the brain and the neck regions. 
image

In order correct for this issue, we need to use  fsl `robustfov`(https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/InitialProcessing) before using `BET`. 
For example, using `robustfov` in the case above, we get the following output image without the neck regions. 
image

### To reproduce

Only happens for some subjects. To reproduce this issue use fsl `BET` for brain extraction. The brain mask generated will also contain the neck regions in addition to the brain. 

### Preconfig

- [X] default
- [X] abcd-options
- [ ] anat-only
- [ ] blank
- [ ] ccs-options
- [ ] fmriprep-options
- [ ] fx-options
- [ ] monkey
- [ ] monkey-ABCD
- [ ] ndmg
- [ ] nhp-macaque
- [ ] preproc
- [ ] rbc-options
- [ ] rodent

### Custom pipeline configuration

_No response_

### Run command

_No response_

### Expected behavior

The output of fsl BET should  only include the brain without other regions of the head (neck, skull). 

### Acceptance criteria

fsl BET should extract only the brain and should be able to remove all the other additional regions of the head such as the skull and the neck for all subjects. 

### Screenshots

_No response_

### C-PAC version

_No response_

### Container platform

_No response_

### Docker and/or Singularity version(s)

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by birajstha**
## Fixes #1928

Fixes #1928 by @birajstha 

## Description

This PR further expands the work of former cpac developer @tergeorge, by addressing the bug on implementation of RobustFOV.
RobustFOV crops the Nifti Image and hence when BET generates the mask, the mask doesnot match the dimension of the original resampled image thus giving error in "3Dcalc" down the pipeline.

## Tests

Tested on few (4-5) sample datasets with and without neck regions in the image where it performed as expected.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `add_fslrobustfov` branch of the repository. 
- [x] My commit messages follow best practices.
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I updated the changelog.
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: CPAC/anat_preproc/utils.py [hunk 1]
```diff
@@ -3,6 +3,30 @@
 
 from CPAC.pipeline import nipype_pipeline_engine as pe
 
+def pad(cropped_image_path, target_image_path):
+    from numpy import asanyarray, zeros_like
```
Comment (shnizzedy) [inline CPAC/anat_preproc/utils.py:7]: Can you [add a docstring](https://github.com/FCP-INDI/.github/blob/main/CONTRIBUTING.md#code-commenting) for this function? I can tell from the PR what is happening here, but if I was just looking in the code, I think I'd need to find where it's called to know what it does

File: CPAC/anat_preproc/utils.py [hunk 2]
```diff
@@ -3,6 +3,30 @@
...
+    from os import path, getcwd
+
+    cropped_image = asanyarray(load(cropped_image_path).dataobj)
+    target_image = asanyarray(load(target_image_path).dataobj)
+
+    # Taking 1 slice to calculate the z dimension shift from top
+    row =target_image.shape[0]//2
+    column= target_image.shape[1]//2
+    a = cropped_image[row, column, :]
+    b = target_image[row, column, :]
```
Comment (shnizzedy) [inline CPAC/anat_preproc/utils.py:18]: I'd [annotate](https://peps.python.org/pep-0526/) most to all of the variables here, but especially ones called `a` and `b` (or give `a` and `b` more descriptive names)

File: CPAC/anat_preproc/utils.py [hunk 3]
```diff
@@ -3,6 +3,30 @@
...
+
+    for z_shift in range(len(b) - len(a) + 1):
+        if (b[z_shift:z_shift+len(a)] == a).all():
+            break
```
Comment (shnizzedy) [inline CPAC/anat_preproc/utils.py:22]: What's this loop for?

**Review by shnizzedy — CHANGES_REQUESTED** [review]
This looks good!

There's some atypical formatting, but that'll get resolved by #1882, so no need to be pedantic about it here.

I do think we should include a docstring for [the new `pad` function](https://github.com/FCP-INDI/C-PAC/blob/b301e24341830660b657b04f729b5b3680f971ba/CPAC/anat_preproc/utils.py#L6-L29) before we merge, though.

**Comment by shnizzedy** [comment]
I rebased both sides of this PR so this PR will include just the relevant commits and won't include all the changes in `develop` since [last March when this branch was originally created from `main`](https://github.com/FCP-INDI/C-PAC/commit/7f94c3a42f0c3eb7c76b15ac5591545d20e8a52f), and I went ahead and did a [conflict resolution](https://github.com/FCP-INDI/C-PAC/pull/2035/commits/5aba132de920a40f047e46c79975c2b8b11c0faf) on the resulting comparison

File: CPAC/anat_preproc/utils.py (hunk 1 above)
Comment (birajstha) [reply CPAC/anat_preproc/utils.py:7]: Thanks for comment. Added now

File: CPAC/anat_preproc/utils.py (hunk 2 above)
Comment (birajstha) [reply CPAC/anat_preproc/utils.py]: Thanks, Added now

File: CPAC/anat_preproc/utils.py (hunk 3 above)
Comment (shnizzedy) [reply CPAC/anat_preproc/utils.py]: Ah, got it. I think making a separate function might be more Pythonic, but it's totally fine here if we don't need to repeat it elsewhere

**Pull request merged**